### DISCUSSION TITLE
emqx_auth_jwt how to verfiy exp iat?

### DISCUSSION THREAD

**Discussion by panla** [emqx/emqx#8319] [Q&A] [answered]
### env

- version: 4.3.15

`emqx_auth_jwt.conf`

```text
auth.jwt.verify_claims = on
auth.jwt.verify_claims.username = %u
```

### token

Python: 3.9

```python
from datetime import datetime, timedelta

import jwt

login_time = datetime.now().timestamp()
token_expired = login_time + timedelta(seconds=36000).total_seconds()

payload = {
    "username": "uuid",
    'exp': token_expired,
    'iat': login_time
}

token = jwt.encode(payload, 'emqxsecret', algorithm="HS256")
```

### error 1

if i add `exp` `iat` into payload and generate a token , thers is rc=5 when connect to emqx, error is `login failed for invalid_signature`

### error 2

```python
login_time = datetime.now()
token_expired = login_time + timedelta(seconds=36000)
```

the error is `login failed for not_authorized`

### ok

if remove `exp` `iat`, it is ok.

How to use them?

**Comment by zmstone** [comment, marked as answer]
Hi.
Replying from phone: `iat` is not checked, so we can rule it out.
To find out  why `exp` is causing the error, could you set debug level logging and share the logs?

**Reply by panla** [reply]
thanks, remove `iat`, it`s OK!
